tests/utils/tools.php

```diff
@@ -607,7 +607,7 @@ function raises($function, $type, $infunction = null) {
             }
             printf("OK: Got %s\n", severityToString($e->getSeverity()));
         } else {
-            printf("ALMOST: Got %s - expected %s\n", get_class($e), $exceptionname);
+            printf("ALMOST: Got %s - expected %s\n", get_class($e), ErrorException::class);
```

Comment:
Actually, this is not a complete fix. This `else` condition is also reached if the severity (e.g. notice, warning) is incorrect. I'll revise that in a bit.

Reply:
Yes. In this case, I think we can change the `catch` in L593 to only catch `ErrorException`, so we only have to handle the case where we've caught messages with a different severity.

We should also wrap the `restore_error_handler` call in a `finally` block for the `try` to properly restore the error handler regardless of outcome.

Reply:
Added `finally` with `restore_error_handler`. I also improved output for converting severity to a string and refactored both `raises()` and `throws()` to reduce conditional nesting.